pkg/subctl/cmd/gather/gather.go

```diff
@@ -107,15 +107,16 @@ func gatherData(cluster *cmd.Cluster) bool {
 	if directory == "" {
 		directory = "submariner-" + time.Now().UTC().Format("20060102150405") // submariner-YYYYMMDDHHMMSS
 	}
-
-	if _, err := os.Stat(directory); os.IsNotExist(err) {
-		err := os.MkdirAll(directory, 0o700)
+	// concatenate the name of the cluster with the root gather directory
+	subDirectory := fmt.Sprintf("%s/%s", directory, cluster.Name)
+	if _, err := os.Stat(subDirectory); os.IsNotExist(err) {
```

Comment:
Why is this checl necessary? Isn't `MkdirAll` capable of ignoring existing directory structure?

> If path is already a directory, MkdirAll does nothing and returns nil.